pallets/ddc-verification/src/lib.rs

```diff
@@ -1309,13 +1360,23 @@ pub mod pallet {
 		}
 
 		fn is_customers_batch_valid(
-			_cluster_id: ClusterId,
-			_era: DdcEra,
+			cluster_id: ClusterId,
+			era: DdcEra,
 			_batch_index: BatchIndex,
 			_payers: &[(T::AccountId, CustomerUsage)],
-			_adjacent_hashes: &[ActivityHash],
+			proof: MerkleProof<ActivityHash, MergeActivityHash>,
+			leaf_with_position: Vec<(u64, ActivityHash)>,
 		) -> bool {
```

Comment:
you need build a tree out of _payers and check that its root is equal to leaf_with_position
leaf_with_position should be a single object - not vector
b

Reply:
can the batch_index be the leaf_position?